src/app/clusters/window-covering-server/window-covering-server.cpp

```diff
@@ -0,0 +1,651 @@
+ * @brief Routines for the Window Covering Server cluster
+ *******************************************************************************
+ ******************************************************************************/
+
+#include "window-covering-server.h"
+
+#include <app/util/af.h>
+
+#include <app/common/gen/attribute-id.h>
+#include <app/common/gen/cluster-id.h>
+#include <app/common/gen/command-id.h>
+
+#include <app/CommandHandler.h>
+#include <app/reporting/reporting.h>
+#include <app/util/af-event.h>
+#include <app/util/af-types.h>
+#include <app/util/attribute-storage.h>
+
+#ifdef EMBER_AF_PLUGIN_SCENES
+#include <app/clusters/scenes/scenes.h>
+#endif // EMBER_AF_PLUGIN_SCENES
+
+using namespace chip;
+
+#define CHECK_BOUNDS_INVALID(MIN, VAL, MAX) ((VAL < MIN) || (VAL > MAX))
+#define CHECK_BOUNDS_VALID(MIN, VAL, MAX) (!CHECK_BOUNDS_INVALID(MIN, VAL, MAX))
+
+#define WC_PERCENTAGE_COEF 100
+#define WC_PERCENT100THS_MAX 10000
+
+static EmberAfStatus writeAttribute(chip::EndpointId ep, chip::AttributeId attributeID, uint8_t * dataPtr,
+                                    EmberAfAttributeType dataType)
+{
+    if (!dataPtr)
+        return EMBER_ZCL_STATUS_INVALID_FIELD;
+
+    EmberAfStatus status =
+        emberAfWriteAttribute(ep, ZCL_WINDOW_COVERING_CLUSTER_ID, attributeID, CLUSTER_MASK_SERVER, dataPtr, dataType);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfWindowCoveringClusterPrint("Err: WC Writing Attribute failed, ep:%u, attr:%04x, err:%04x", ep, attributeID, status);
+    }
+    return status;
+}
+
+static EmberAfStatus readAttribute(chip::EndpointId ep, chip::AttributeId attributeID, uint8_t * dataPtr, uint16_t readLength)
+{
+    if (!dataPtr)
+        return EMBER_ZCL_STATUS_INVALID_FIELD;
+
+    EmberAfStatus status =
+        emberAfReadAttribute(ep, ZCL_WINDOW_COVERING_CLUSTER_ID, attributeID, CLUSTER_MASK_SERVER, dataPtr, readLength, NULL);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        emberAfWindowCoveringClusterPrint("Err: WC Reading Attribute failed, ep:%u, attr:%04x, err:%04x", ep, attributeID, status);
+    }
+
+    return status;
+}
+
+//------------------------------------------------------------------------------
+// WindowCover
+//------------------------------------------------------------------------------
+
+WindowCover WindowCover::sInstance;
+
+WindowCover & WindowCover::Instance()
+{
+    return sInstance;
+}
+
+WindowCover::WindowCover() {}
+
+void WindowCover::TypeSet(EmberAfWcType type)
+{
+    writeAttribute(mEndPoint, ZCL_WC_TYPE_ATTRIBUTE_ID, (uint8_t *) &type, ZCL_ENUM8_ATTRIBUTE_TYPE);
+}
+
+EmberAfWcType WindowCover::TypeGet(void)
+{
+    EmberAfWcType type;
+    readAttribute(mEndPoint, ZCL_WC_TYPE_ATTRIBUTE_ID, (uint8_t *) &type, sizeof(uint8_t));
+    return type;
+}
+
+void WindowCover::ConfigStatusSet(WindowCover::ConfigStatus status)
+{
+    writeAttribute(mEndPoint, ZCL_WC_CONFIG_STATUS_ATTRIBUTE_ID, (uint8_t *) &status, ZCL_BITMAP8_ATTRIBUTE_TYPE);
```

Comment:
You might also want to use the functions provided by `#include <app/common/gen/attributes/Accessors.h>`

Reply:
Will be updated the next PR after this one , when @ricardo-casallas did this work it , these were not available

Reply:
I have replaced all the writeAttribute and readAttribute calls with their corresponding Accessors.h functions.